### PR TITLE
feat: add dedicated horizontal layout for Power Source

### LINKED ISSUE #2493: Power Source: add horizontal layout
**Actual Behaviour**

Changing the orientation to landscape, the horizontal layout is used, but it is scaled to the higher width which looks strange.

**Expected Behaviour**

The app has an extra layout for this screen in landscape orientation which looks like this:

`+----------+----------+`
`|   PV1    |    PV2   |`
`+----------+----------+`
`|   PV3    |    PCS   |`
`+----------+----------+`

**Steps to reproduce it**

1. Open app
2. Open Power Source
3. Change orientation to landscape

**Screenshots of the issue**

Screenshot_20240618-232530

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by marcnause**
Fixes #2493

## Changes 
- Add horizontal layout for Power Source
- Add new `dimen` for new layout

## Screenshots / Recordings  
Screenshot_20240625-231244

**Checklist**: 
- [x] **No hard coding**: I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard coding any value.
- [x] **No end of file edits**: No modifications done at end of resource files `strings.xml`, `dimens.xml` or `colors.xml`.
- [x] **Code reformatting**: I have reformatted code and fixed indentation in every file included in this pull request.
- [x] **No extra space**: My code does not contain any extra lines or extra spaces than the ones that are necessary.

### CODE REVIEW

**Comment by AsCress** [comment]
@marcnause The layout looks fantastic !! 
The only issue I can see in here is that the _UP_ arrow button decreases the voltage whereas the _DOWN_ arrow button increases the voltage 🙂😂😂.
I am attaching a video for your reference.

https://github.com/fossasia/pslab-android/assets/125425881/10fdac9f-95b4-4aae-8ffa-baf127d7f32e

Perhaps we can also fix some hardcodes in a line or two. I'll just mark those areas in a review for your reference. I also tend to forget about a lot about the hardcoded values when dealing with _0dp_ margins :)).

File: app/src/main/res/layout-land/activity_powersource_layout.xml [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+            android:orientation="vertical">
+
+            <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:baselineAligned="false"
+                android:orientation="horizontal">
+
+                <RelativeLayout
+                    android:layout_width="0dp"
```
Comment (AsCress) [inline app/src/main/res/layout-land/activity_powersource_layout.xml]: Hardcoded

File: app/src/main/res/layout-land/activity_powersource_layout.xml [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+            </LinearLayout>
+
+            <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:baselineAligned="false"
+                android:orientation="horizontal">
+
+                <RelativeLayout
+                    android:layout_width="0dp"
```
Comment (AsCress) [inline app/src/main/res/layout-land/activity_powersource_layout.xml]: Hardcoded

File: app/src/main/res/layout-land/activity_powersource_layout.xml [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+                                app:progress_primary_color="@color/indicator_color"
+                                app:progress_secondary_color="@color/progress_secondary_color"
+                                app:start_offset="45" />
+                        </LinearLayout>
+                    </androidx.cardview.widget.CardView>
+
+                </RelativeLayout>
+
+                <RelativeLayout
+                    android:layout_width="0dp"
```
Comment (AsCress) [inline app/src/main/res/layout-land/activity_powersource_layout.xml]: Hardcoded

File: app/src/main/res/layout-land/activity_powersource_layout.xml (hunk 1 above)
Comment (marcnause) [reply app/src/main/res/layout-land/activity_powersource_layout.xml]: I have replaced "0dp" with "@dimen/weight_scale".

File: app/src/main/res/layout-land/activity_powersource_layout.xml (hunk 2 above)
Comment (marcnause) [reply app/src/main/res/layout-land/activity_powersource_layout.xml]: I have replaced "0dp" with "@dimen/weight_scale".

File: app/src/main/res/layout-land/activity_powersource_layout.xml (hunk 3 above)
Comment (marcnause) [reply app/src/main/res/layout-land/activity_powersource_layout.xml]: I have replaced "0dp" with "@dimen/weight_scale".

**Comment by marcnause** [comment]
I checked the buttons and the up-button incremented the values and the down-button decremented the value. I think the problem was that the up-button was on the left and the down-button on the right. To me this does not feel natural. I checked [my alarm clock](https://de.hama.com/00186320/hama-funkwecker-rc-550-mit-nachtlicht-funktion) 😂 and it has the buttons the other way around. I suspect that cultures which write from right to left may expect the button order the way it was, but I we don't support right to left layouts at the moment. I changed the order and it feels more natural to me now. @AsCress, does it feel right for you now too? 

I also did some minor changes in the vertical layout to make it a little more consistent with the horizontal version. I think the sizes of the elements are aligned a little better now.

**Review by CloudyPadmal — APPROVED** [review]

File: app/src/main/res/layout/activity_powersource_layout.xml [hunk 4]
```diff
@@ -66,25 +66,25 @@
                             android:orientation="horizontal">
```
Comment (AsCress) [inline app/src/main/res/layout/activity_powersource_layout.xml]: @marcnause The layout looks fantastic and works really well in the horizontal orientation. However, in the vertical orientation, the size of the buttons becomes too big, due to which the text in the _EditTexts_ cuts off. I'm attaching a screen recording for your reference.

https://github.com/fossasia/pslab-android/assets/125425881/525dd764-edc7-4d57-9a66-a7d401abff6b

Perhaps we can retain the older configuration of the vertical layout and just make changes in the horizontal one ? Let me know how this sounds to you.

**Review by AsCress — CHANGES_REQUESTED** [review]

**Comment by marcnause** [comment]
OK, one last try before I replace the vertical layout with the original one.

I made recordings with my test devices:

###  Yoga Smart Tab
https://github.com/fossasia/pslab-android/assets/11596220/508d9cca-8c38-4452-b7a1-a3d1e5823ab4

### Fairphone 4
https://github.com/fossasia/pslab-android/assets/11596220/c4419bda-e175-4dc3-b847-8b814f49846b

### Motorola Moto X Play
https://github.com/fossasia/pslab-android/assets/11596220/c95506d8-9baa-4ba1-b54f-029cbb7714ab

On the tablet the buttons of the vertical layout are huge, but I don't have any idea for a better vertical layout for large screens at the moment.

On the Moto X Play the unit of the PCS is cut off. We could create an extra horizontal layout for small screens, but the more layouts we have, the harder it is to maintain them. Any thoughts on this?

File: app/src/main/res/layout/activity_powersource_layout.xml (hunk 4 above)
Comment (marcnause) [reply app/src/main/res/layout/activity_powersource_layout.xml]: OK, one last try before I replace the vertical layout with the original one.

I made recordings with my test devices:

###  Yoga Smart Tab
https://github.com/fossasia/pslab-android/assets/11596220/508d9cca-8c38-4452-b7a1-a3d1e5823ab4

### Fairphone 4
https://github.com/fossasia/pslab-android/assets/11596220/c4419bda-e175-4dc3-b847-8b814f49846b

### Motorola Moto X Play
https://github.com/fossasia/pslab-android/assets/11596220/c95506d8-9baa-4ba1-b54f-029cbb7714ab

On the tablet the buttons of the vertical layout are huge, but I don't have any idea for a better vertical layout for large screens at the moment.

On the Moto X Play the unit of the PCS is cut off. We could create an extra horizontal layout for small screens, but the more layouts we have, the harder it is to maintain them. Any thoughts on this?

**Comment by AsCress** [comment]
@marcnause I have updated the _PowerSourceActivity_ layouts, to use a more flexible `ConstraintLayout` for the inputs. I have also corrected some dimensions here and there.
Could you please test it on the devices available with you and share your thoughts ?

**Comment by marcnause** [comment]
@AsCress I tried looking at he current status of the layout, but the app crashes on all of my devices when I open the Power Source screen. 

edit: The crash seems to be a problem which is not related to this PR. I have created a new issue: https://github.com/fossasia/pslab-android/issues/2519

**Comment by marcnause** [comment]
Here are the screenshots of the Power Source screens. What' strange is that the unit of PCS has changed from "mA" to "m".

## Fairphone 4 (vertical)


## Fairphone 4 (horizontal)


## Moto X Play (vertical)


## Moto X play (horizontal)


## Yoga Smart Tab (vertical)


## Yoga Smart Tab (horizontal)

**Comment by AsCress** [comment]
@marcnause I see. The issue is `android:maxLength="7"` being set for all the `EditTexts`. I don't think we need a `maxLength` for the `EditTexts` we have in there. I'll just correct this. Also, I recognize that on some devices with smaller screen sizes (such as the Moto X Play), the units of voltage and current may not be properly displayed when higher values are chosen. But I think that it shouldn't be much of a problem. What are your views on this ?

**Comment by marcnause** [comment]
The issue with the units being cut off on smaller screens already existed previous to this PR. I will add a new issue for that.

**Pull request merged**